### DISCUSSION TITLE
Module parse failed for loader

### DISCUSSION THREAD

**Discussion by ezno12** [tauri-apps/tauri#6808] [Q&A]
I'm working on code base react, node, electron. my objective to migrate the project to tauri. when I work with fs tauri api i faced problem of processing loader.

This the bug I had:
![tauri](https://user-images.githubusercontent.com/39200164/234840971-79f601f9-de62-4a54-bd58-a734a6c940e3.PNG)

This ts Config file:
![tsconfig](https://user-images.githubusercontent.com/39200164/234841025-6264a722-2be0-4532-8f1c-80094d49c9e3.PNG)

This Tauri config file:
![tauri config](https://user-images.githubusercontent.com/39200164/234841069-2858683d-99f0-4d0e-86db-7b3e0e8f34e5.PNG)

Reproduction

    installed all prerequest of tauri
    Added tauri api package
    yarn tauri dev
    add import statement import { BaseDirectory, exists } from "@tauri-apps/api/fs"
    add code to check image exsits in picture windows folder : await exists("avatar.png", { dir: BaseDirectory.Picture })

Expected behavior

to return Promise as tauri documentation said.

tauri dev info:

Environment
  › OS: Windows 10.0.19045 X64
  › Webview2: 112.0.1722.58
  › MSVC:
      - Visual Studio Build Tools 2022
  › Node.js: 14.18.3
  › npm: 6.14.15
  › pnpm: Not installed!
  › yarn: 1.22.19
  › rustup: 1.25.2
  › rustc: 1.69.0
  › cargo: 1.69.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc

Packages
  › @tauri-apps/cli [NPM]: 1.2.3
  › @tauri-apps/api [NPM]: 1.2.0
  › tauri [RUST]: 1.2.4,
  › tauri-build [RUST]: 1.2.1,
  › tao [RUST]: 0.15.8,
  › wry [RUST]: 0.23.4,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../build
  › devPath: http://localhost:3000/
  › framework: React

App directory structure
  ├─ .git
  ├─ .husky
  ├─ .vscode
  ├─ electron
  ├─ node_modules
  ├─ public
  ├─ scripts
  ├─ src
  └─ src-tauri

**Comment by HuakunShen** [comment]
I can't reproduce this with Tauri's react template.

**Reply by ezno12** [reply]
I'm looking to migrate from electron to tauri on existing project ! I can't start with a tauri's react template. I think the problem produce it from webpack.

**Comment by simonhyll** [comment]
All I can see that might be wrong is that your Node.js version is a bit outdated along with some dependency versions, so I would start there.

Other than that just doing the steps you described hasn't caused any issues for me, so the only way for me to give better support with this is if you create a minimal reproduction for it that I can clone and try out.

But yea this sort of issue is usually related to outdated versions, or an incorrectly set up bundler.
